Automattic/wp-calypso
# Reader: fix follow count for Tracks events

#### Changes proposed in this Pull Request

In the pre-Redux days, Tracks events sent by Reader used to send the current user's follow count as an event property. This has not been working since 2017. As the code notes, the plan was to move this to Redux middleware:

https://github.com/Automattic/wp-calypso/blob/61acc957638fc7009632af4c0c058aa6bc0e82b7/client/reader/stats.js#L106-L109

The corresponding issue is https://github.com/Automattic/wp-calypso/issues/14470, and the missing count has recently been brought up by @martinremy in in p5PDj3-4WA-p2.

#### Testing instructions


Turn on Analytics logging in your console with:

`localStorage.setItem('debug', 'calypso:analytics*');`

Go to http://calypso.localhost:3000/read/conversations and try unfollowing and following an existing conversation via the ellipsis menu. The resulting Tracks events should contain a `subscription_count` key:

<img width="968" alt="Screen Shot 2020-11-27 at 14 48 38" src="https://user-images.githubusercontent.com/17325/100401873-aa5c6200-30bf-11eb-8460-249bbd072415.png">

Partly fixes #14470 (we'll need to migrate all of the existing `recordTrack()` uses in a followup PR).
